packages/compiler-cli/src/ngcc/src/rendering/esm_renderer.ts

```diff
@@ -27,6 +30,23 @@ export class EsmRenderer extends Renderer {
     imports.forEach(i => { output.appendLeft(0, `import * as ${i.as} from '${i.name}';\n`); });
   }
 
+  addExports(
+      output: MagicString, entryPointBasePath: string, privateExports: ExportInfo[],
+      useDts: boolean): void {
+    privateExports.forEach(privateExport => {
+      const fromPath = useDts ? privateExport.dtsFrom : privateExport.from;
+      if (!fromPath) {
+        throw new Error(
+            `There is no typings path for ${privateExport.identifier} in ${privateExport.from}`);
+      }
+      const basePath = fromPath.replace(/\.(js|d\.ts$)/, '');
```

Comment:
Should include `$` for js as well:

```suggestion
      const basePath = fromPath.replace(/\.(js|d\.ts)$/, '');
```